adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -379,17 +409,20 @@ def connect(self, clean_session=True):
                 return result
 
     def disconnect(self):
-        """Disconnects the MiniMQTT client from the MQTT broker.
-        """
+        """Disconnects the MiniMQTT client from the MQTT broker."""
         self.is_connected()
         if self.logger is not None:
             self.logger.debug("Sending DISCONNECT packet to broker")
-        self._sock.send(MQTT_DISCONNECT)
+        try:
+            self._sock.send(MQTT_DISCONNECT)
+        except RuntimeError as e:
+            if self.logger:
+                self.logger.warning("Unable to send DISCONNECT packet: {}".format(e))
         if self.logger is not None:
             self.logger.debug("Closing socket")
         self._sock.close()
         self._is_connected = False
-        self._subscribed_topics = None
+        self._subscribed_topics = []
```

Comment:
May want to check for `len(_subscribed_topics)` first, then empty it?

Reply:
Maybe so, but what is important is that we have a list here. 
Otherwise we will get something like this when trying to connect later on:
```
  File "code.py", line 287, in <module>
  File "/lib/adafruit_minimqtt/adafruit_minimqtt.py", line 406, in connect
  File "code.py", line 137, in connect
  File "/lib/adafruit_minimqtt/adafruit_minimqtt.py", line 628, in subscribe
AttributeError: 'NoneType' object has no attribute 'append'

Press any key to enter the REPL. Use CTRL-D to reload.
```
Do you mind making that shange as part of another PR?

Reply:
> Do you mind making that change as part of another PR?

I think it should be folded into this PR since we're introducing a list here and want to handle the edge case.

Reply:
> > Do you mind making that change as part of another PR?
> 
> I think it should be folded into this PR since we're introducing a list here and want to handle the edge case.

Oh... that is not new. If you look at:  https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/blob/e02d658425f099c5a573cd0e0c2b3b4150c25cfd/adafruit_minimqtt/adafruit_minimqtt.py#L162  before this PR, you can see that this was always a list. The disconnect code is the 'bug' that was improperly setting it to None instead of [].

Reply:
@brentru if you would like, I can undo this change on this PR and open a separate issue for specifically dealing with the disconnect() / connect() bug. LMK, pls.